From 04521b987026b8a00f9bc784a63daff3af31e0cb Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Tue, 18 Jan 2022 13:06:06 +0100
Subject: [PATCH 43/86] linux: Fix warning when using fake UPS

When the battery level is refreshed on a fake UPS:
(upowerd:320760): UPower-Linux-WARNING **: 12:59:26.205: incomplete read (0<8)
---
 src/linux/up-device-hid.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/linux/up-device-hid.c b/src/linux/up-device-hid.c
index 6c84801..e87ccf6 100644
--- a/src/linux/up-device-hid.c
+++ b/src/linux/up-device-hid.c
@@ -395,6 +395,9 @@ up_device_hid_refresh (UpDevice *device)
 	int rd;
 	UpDeviceHid *hid = UP_DEVICE_HID (device);
 
+	if (hid->priv->fake_device)
+		goto update_time;
+
 	/* read any data */
 	rd = read (hid->priv->fd, ev, sizeof (ev));
 
@@ -423,6 +426,7 @@ up_device_hid_refresh (UpDevice *device)
 	/* fix up device states */
 	up_device_hid_fixup_state (device);
 
+update_time:
 	/* reset time */
 	g_object_set (device, "update-time", (guint64) g_get_real_time () / G_USEC_PER_SEC, NULL);
 out:
-- 
2.35.1

